src/test/java/com/puppycrawl/tools/checkstyle/TreeWalkerTest.java

```diff
@@ -253,6 +262,21 @@ public void testRequiredTokenIsEmptyIntArray() throws Exception {
         }
     }
 
+    @Test
+    public void testBehaviourWithZeroChecks() throws Exception {
+        final TreeWalker treeWalker = new TreeWalker();
+        final String errMsg = "There should be no parsing or walking ASTs if there are no checks";
+        mockStatic(TreeWalker.class);
+        when(TreeWalker.parse(any(FileContents.class)))
+                .thenThrow(new IllegalStateException(errMsg));
```

Comment:
Yes it can be done like that as well. But again we will have to catch the runtimeexception thrown by the `verify` method in there and again fail the test !